reana_server/status.py

```diff
@@ -341,12 +361,285 @@ def get_status(self):
+                )
+                for status in self.statuses
+            }
+            for compute_backend in self.compute_backends
+        }
+
+        job_statuses["kubernetes_api"] = {
+            "running": len(self.get_k8s_jobs_by_status("Running")),
+            "pending": len(self.get_k8s_jobs_by_status("Pending")),
+        }
+        return job_statuses
+
+
+class ClusterHealthStatus(enum.Enum):
+    """Enumeration of cluster health statuses."""
+
+    healthy = 0
+    warning = 1
+    critical = 2
+
+
+class ClusterHealth:
+    """Class to retrieve REANA cluster health information."""
+
+    def __init__(self):
+        """Initialise Cluster Health class."""
+        self.node = self.get_node_health()
+        self.job = self.get_job_health()
+        self.workflow = self.get_workflow_health()
+        self.session = self.get_session_health()
+
+    @staticmethod
+    def get_health_status(percentage):
+        """Calculate quota health status."""
+        health = ClusterHealthStatus.healthy
+        if percentage >= 50:
+            if percentage >= 75:
+                health = ClusterHealthStatus.critical
+            else:
+                health = ClusterHealthStatus.warning
+        return health.name
+
+    @staticmethod
+    def get_percentage(amount, total):
+        """Get percentage value based on ``amount`` and ``total``."""
+        return round((amount / total if total else 0) * 100)
+
+    def get_node_health(self):
+        """Get cluster nodes health information."""
+        nodes_status_obj = NodesStatus()
+
+        total = len(nodes_status_obj.get_nodes())
+        unschedulable = len(nodes_status_obj.get_unschedulable_nodes())
+        percentage = ClusterHealth.get_percentage(unschedulable, total)
+
+        return {
+            "available": total - unschedulable,
+            "unschedulable": unschedulable,
+            "percentage": percentage,
+            "health": ClusterHealthStatus.critical.name
+            if unschedulable > 0
+            else ClusterHealthStatus.healthy.name,
+            "sort": 0,
+        }
+
+    def get_job_health(self):
+        """Get cluster jobs health information."""
+        jobs_status_obj = JobsStatus()
+
+        running = len(jobs_status_obj.get_k8s_jobs_by_status("Running"))
+        pending = len(jobs_status_obj.get_k8s_jobs_by_status("Pending"))
+        total = running + pending
+        percentage = ClusterHealth.get_percentage(pending, total)
+
+        return {
+            "running": running,
+            "pending": pending,
+            "percentage": percentage,
+            "health": ClusterHealth.get_health_status(percentage),
+            "sort": 1,
+        }
+
+    def get_workflow_health(self):
+        """Get cluster workflows health information."""
+        wf_status_obj = WorkflowsStatus()
+
+        running = wf_status_obj.get_workflows_by_status(RunStatus.running)
+        queued = wf_status_obj.get_workflows_by_status(RunStatus.queued)
```

Comment:
Question: why we are ignoring `pending` workflows? Since workflows can get stuck in `pending` state for some time, it could be important while checking the cluster health

Reply:
Makes sense! Added.